File: PYTHON/Program/Prodi.py
from Course import Course
from Dosen import Dosen
from Mahasiswa import Mahasiswa

class Prodi:
    __nama_prodi = ""
    __kode = ""
    __course = Course()
    __mahasiswa = Mahasiswa()
    __dosen = Dosen()

    def __init__(self, nama_prodi = "", kode = "", course = Course(), dosen = Dosen(), mahasiswa = Mahasiswa()):
        # Set attributes pada class mahasiswa
        self.__nama_prodi = nama_prodi
        self.__kode = kode
        self.__course = course
        self.__dosen = dosen
        self.__mahasiswa = mahasiswa

    # Getter dan Setter

    # Getter dan setter untuk nama mahasiswa
    def get_nama_prodi(self):
        return self.__nama_prodi
    def set_nama_prodi(self, nama_prodi):
        self.__nama_prodi = nama_prodi

    def get_kode(self):
        return self.__kode
    def set_kode(self, kode):
        self.__kode = kode

    def get_course(self):
        return self.__course
    def set_course(self, course):
        self.__course = course

    def get_dosen(self):
            return self.__dosen
    def set_dosen(self, dosen):
        self.__dosen = dosen
    
    def get_mahasiswa(self):
            return self.__mahasiswa
    def set_course(self, mahasiswa):
        self.__mahasiswa = mahasiswa

File: CPP/Program/Prodi.cpp
#include "Header.hh"

Prodi :: Prodi()
{
    nama_prodi = "";
    kode = "";
}

Prodi :: Prodi(string nama_prodi, string kode, Course course, Dosen dosen, Mahasiswa mahasiswa)
{
    this->nama_prodi = nama_prodi;
    this->kode = kode;
    this->course = course;
    this->dosen = dosen;
    this->mahasiswa = mahasiswa;
}

string Prodi::getNama_prodi()
{
    return nama_prodi;
}

void Prodi ::setNama_prodi(string nama_prodi)
{
    this->nama_prodi = nama_prodi;
}

string Prodi::getKode()
{
    return kode;
}

void Prodi ::setKode(string kode)
{
    this->kode = kode;
}

Course Prodi::getCourse()
{
    return course;
}

void Prodi ::setCourse(Course course)
{
    this->course = course;
}

Dosen Prodi::getDosen()
{
    return dosen;
}

void Prodi ::setDosen(Dosen dosen)
{
    this->dosen = dosen;
}

Mahasiswa Prodi::getMahasiswa()
{
    return mahasiswa;
}

void Prodi ::setMahasiswa(Mahasiswa mahasiswa)
{
    this->mahasiswa = mahasiswa;
}

Prodi::~Prodi()
{

}